src/main/java/CommandParser.java

```diff
@@ -0,0 +1,21 @@
+public class CommandParser {
+    private final TaskList taskList;
+
+
+    public CommandParser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    public void parseCommand(String userInput) throws HandleException {
+        if (userInput.trim().isEmpty()) {
+            throw new HandleException("The input cannot be empty!");
+        } else if (userInput.equalsIgnoreCase("list")) {
+            taskList.listTasks();
+        } else if (!userInput.startsWith("todo") && !userInput.startsWith("deadline") && !userInput.startsWith("event")) {
```

Comment:
This is a fairly complex statement. You might want to break this into sub-expressions and store values into some boolean variables to check for the "if" condition.